### PR TITLE
Fix sign-in while running a server from amo:dev

### LINKED ISSUE #2941: Cannot login with the dev server
Hi!

### Describe the problem and steps to reproduce it:

It looks like the dev server has an issue to authenticate users. 

1. run `yarn amo:dev`
2. go to the web interface and click "Log in/Sign up"
3. authenticate (maybe create an account first)
4. observe a 404 Page not found, without being authenticated

### What happened?

404 Page not found. Below is the console output:

```
index.js:57 [amo] Skipping componentWillMount() because visibleAddonType is invalid: fxa-authenticate
index.js:180 [amo] Rendering 404 because visibleAddonType is invalid: fxa-authenticate
```

### What did you expect to happen?

Being authenticated.

### PR DESCRIPTION
**Pull request by kumar303**
Fixes #2941 

This does a few things:
* Both `amo:dev` and `amo:stage` go through the proxy which means both the frontend site and the backend API are served on the same domain like production. This should help catch same origin issues locally.
* Sign-ins work on `amo:dev`
* Sign-ins do not work on `amo:stage` (like before) because we need to configure a new FxA client. I'll do that in a follow-up.

### CODE REVIEW

**Comment by willdurand** [comment]
🎉  💯 

Just tested this PR and it works well. Thanks @kumar303. It also works for newly created accounts (which did not work either before), except that the user gets redirected to `http://localhost:3000/en-US/firefox/users/edit`, which does not exit (404). Logout works too.

**Comment by kumar303** [comment]
@willdurand great, thanks for testing it out. I am reminded now that the redirection 404 (for new accounts) is a known issue that will be fixed when https://github.com/mozilla/addons-frontend/issues/2781 lands. However, with my patch you will be logged in when you type in the URL `localhost:3000` after the 404 so it's not too annoying.

**Comment by tofumatt** [comment]
Just testing this now, I was beat up by a migraine most of today, sorry! 😢

File: package.json [hunk 1]
```diff
@@ -66,10 +67,11 @@
       }
     },
     "amo:stage": {
-      "command": "better-npm-run start-dev",
+      "command": "better-npm-run start-dev-proxy",
```
Comment (tofumatt) [inline package.json]: Shoot! My bad, I think this change was my fault 😢

**Comment by kumar303** [comment]
btw, I also tested add-on/theme installation through a [custom Nightly build](https://github.com/mozilla/addons-frontend/blob/master/docs/moz-addon-manager.md#developing-with-a-local-server)

**Comment by kumar303** [comment]
@willdurand be sure to clear your cookies when switching between `yarn amo:dev`, `yarn amo`, and `yarn amo:stage` since they all share the same domain (`localhost:3000`).

**Comment by willdurand** [comment]
ah good to know, thanks!

**Pull request merged**